### DISCUSSION TITLE
Getting segmentation mask of PSM arms + suture from AMBF

### DISCUSSION THREAD

**Discussion by sunshineinsandiego** [WPI-AIM/ambf#207] [General]
Hi - I was reading [here](https://github.com/WPI-AIM/ambf/wiki/preprocessing-shaders) but I'm not sure how to go about recovering segmented images. 

Ideally I would like to be able to recover the segmentation masks of the PSM arms and the suture needle, which appears to be possible, but advice on how to do so would be most appreciated. Thank you!

**Comment by adnanmunawar** [comment]
Hello,

We have a camera for the surgical robotics challenge (SRC) that implements this.

https://github.com/collaborative-robotics/surgical_robotics_challenge/blob/master/ADF/world/world_stereo.yaml#L64

 Once you run the SRC you can subscribe to the topic '/ambf/env/cameras/cameraL2/ImageData' to get the segmented mask. You can do this for any camera in AMBF as suggested by the link that you pointed to.

![segmentation](https://github.com/WPI-AIM/ambf/assets/5005445/b4cedaec-b0e4-4e70-9376-54a1574bca04)

**Comment by sunshineinsandiego** [comment]
Will do, thank you as usual for the help!
